Sources/eppo/UniversalFlagConfig.swift

```diff
@@ -0,0 +1,159 @@
+import Foundation
+
+public struct UniversalFlagConfig : Decodable {
+    let createdAt: Date?;
+    let flags: [String: UFC_Flag];
+    // todo: add bandits
+    
+    static func decodeFromJSON(from jsonString: String) throws -> UniversalFlagConfig {
+        guard let jsonData = jsonString.data(using: .utf8) else {
+            throw UniversalFlagConfigError.notUTF8Encoded("Failed to encode JSON string into UTF-8 data.")
+        }
+        
+        let decoder = JSONDecoder()
+        
+        // Dates could be in base64 encoded format or not
+        decoder.dateDecodingStrategy = .custom { decoder -> Date in
+            let container = try decoder.singleValueContainer()
+            let dateStr = try container.decode(String.self)
+            guard let date = parseUtcISODateElement(dateStr) else {
+                print(dateStr)
+                throw DecodingError.dataCorruptedError(in: container, debugDescription: "Invalid date format for: <\(dateStr)>")
+            }
+            return date
+        }
+        
+        do {
+            return try decoder.decode(UniversalFlagConfig.self, from: jsonData)
+        } catch let error as DecodingError {
+            switch error {
+            case .typeMismatch(_, let context):
+                throw UniversalFlagConfigError.parsingError("Type mismatch: \(context.debugDescription)")
+            case .valueNotFound(_, let context):
+                throw UniversalFlagConfigError.parsingError("Value not found: \(context.debugDescription)")
+            case .keyNotFound(let key, let context):
+                throw UniversalFlagConfigError.parsingError("Key not found: \(key.stringValue) - \(context.debugDescription)")
+            case .dataCorrupted(let context):
+                throw UniversalFlagConfigError.parsingError("Data corrupted: \(context.debugDescription)")
+            default:
+                throw UniversalFlagConfigError.parsingError("JSON parsing error: \(error.localizedDescription)")
+            }
+        } catch {
+            throw UniversalFlagConfigError.parsingError("Unexpected error: \(error.localizedDescription)")
+        }
+    }
+}
+
+// enums
+enum UniversalFlagConfigError: Error, CustomNSError, LocalizedError {
+    case notUTF8Encoded(String)
+    case parsingError(String)
+    
+    static var errorDomain: String { return "UniversalFlagConfigError" }
+    
+    var errorCode: Int {
+        switch self {
+        case .notUTF8Encoded:
+            return 100
+        case .parsingError:
+            return 101
+        }
+    }
+    
+    var errorDescription: String? {
+        switch self {
+        case .notUTF8Encoded(let message):
+            return message
+        case .parsingError(let message):
+            return message
+        }
+    }
+}
+
+enum UFC_VariationType: String, Decodable {
```

Comment:
curious about your naming decision on the `UFC_` prefix, haven't seen that in other SDKs. could we keep camelCase for consistently?

Reply:
I did this because I had both the old models and the new UFC models co-existing and there was some conflict in the names, such as `Variant`